torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1491,8 +1503,11 @@ void ProcessGroupNCCL::watchdogHandler() {
           timeSinceLastPollStore >= heartbeatTimeoutInSec_ * 1000) {
         lastTimePollStore = currentTime;
         if (store_->check({std::string(TIMEOUT_DUMP)}) && !optAsyncDebugDump) {
+          auto wakeUpTime = std::chrono::steady_clock::now() +
+              std::chrono::milliseconds(waitTimeoutDumpInMilSec_);
           optAsyncDebugDump = launchAsyncDebugDump();
           waitForDumpOrTimeout(*optAsyncDebugDump);
+          extraWaitForDumpUntil(wakeUpTime);
```

Comment:
i’m wondering if it would be better to keep 'extraWaitForDumpUntil' as a separate helper or, maybe pass the ‘wakeup_time’ as a 3rd argument to `waitForDumpOrTimeout`.  It seems like that would satisfy all the cases we have today.

separately, we could make a helper for getWakeupTime(time_interval) that we use to compute `wakeUpTime = std::chrono::steady_clock::now() +
              std::chrono::milliseconds(waitTimeoutDumpInMilSec_)`